### DISCUSSION TITLE
Contract write "missing block number" error

### DISCUSSION THREAD

**Discussion by SpasiboKojima** [wevm/wagmi#1037] [Q&A]
I've been using Wagmi for ~2 months and only recently I started getting instant errors from time to time when submitting transactions.
I'm using Alchemy provider on Polygon Mumbai testnet. Transactions are prepared with `usePrepareContractWrite`.
Worth noting that resending the same request from browsers Network tab results in a successful fetch of transaction receipt.

Error from `useWaitForTransaction` (I removed api key from url):
`Error: processing response error (body="{\"jsonrpc\":\"2.0\",\"id\":47,\"error\":{\"code\":-32000,\"message\":\"missing block number\"}}", error={"code":-32000}, requestBody="{\"method\":\"eth_getTransactionReceipt\",\"params\":[\"0xce017aee1681ec5d855446f3fee393c882e523712e40c4de38313d2b70aea662\"],\"id\":47,\"jsonrpc\":\"2.0\"}", requestMethod="POST", url="https://polygon-mumbai.g.alchemy.com/v2/<API_KEY>", code=SERVER_ERROR, version=web/5.6.1)`

Request body:
`{"method":"eth_getTransactionReceipt","params":["0xbcec71d209e8bf51bb9c3449819e17a161c690acd77cf3224713dd12810b8a16"],"id":48,"jsonrpc":"2.0"}`

Response:
`{"jsonrpc":"2.0","id":48,"error":{"code":-32000,"message":"missing block number"}}`

**Comment by tmm** [comment]
Can you post a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)? Sort of hard to tell what could be happening with the information you provided.

**Reply by SpasiboKojima** [reply]
Thanks for reply, sure. Pretty much this, I suppose ABI and contract address are irrelevant since again, it seems like a `useWaitForTransaction` issue,

```
import { useContractWrite, usePrepareContractWrite, useWaitForTransaction } from 'wagmi';

const BuyTicketsButton = () => {
  const [amount, setAmount] = useState(1);
  const { config, error } = usePrepareContractWrite({
    addressOrName: address,
    contractInterface: RaffleABI,
    functionName: 'purchaseTicket',
    args: [amount],
    overrides: {
      value: amount.toString(),
    },
  });
  const { data, write } = useContractWrite(config);
  const { isLoading } = useWaitForTransaction({
    hash: data?.hash,
  });
  
  return (
      <>
           <input
                value={amount ?? 1}
                onChange={(e) => setAmount(Number(e.target.value))}
                inputMode="numeric"
              />
           <button onClick={write}>
                Buy ticket
          </button>
      </>
  );
};
```
